### ISSUE TITLE
What is the difference between frame00xxx-pose.txt file 4x4 pose matrix and DSO: Direct sparse odometry of 12 values poses.??

### ISSUE THREAD

**Issue by DRAhmadFaraz** [NVlabs/geomapnet#19]
@samarth-robo Sir, I want to ask What is the difference between `frame00xxx-pose.txt `file 4x4 pose matrix for each corresponding image and `DSO: Direct sparse odometry txt files `of 12 values of poses for each corresponding image.??

Does your code needs to compare the results from both of them used as a ground truth.? 
why your code needed 2 poses files for the comparison.? what is the difference between these two files.??

Kindly guide me. I will be thankful to you..
Best regards

**Comment by DRAhmadFaraz** [comment]
@samarth-robo Sir, I am waiting for your response.

Thanx in advance

**Comment by samarth-robo** [comment]
DSO is a visual odometry algorithm, and is used while training MapNet++. The `frame00xxx-pose.txt` contains absolute pose of the algorithm, while the DSO files contain relative camera motion from one frame to the next. The first column in the DSO file is the frame number.
For more details you will have the read our paper and the code. I am not able to give further support on this right now because of time constraints.

**Comment by AntiLibrary5** [comment]
"The frame00xxx-pose.txt contains absolute pose of the algorithm, while the DSO files contain relative camera motion from one frame to the next."
I don't think its the actual camera poses. The authors of 7Scenes dataset note that their labels are: "camera pose matrices H that encode the 3D rotation and translation from camera space to world space", so a camera to world matrix.

And from the dataloader of 7Scences, we take the first 12 elements as input to the pose_utils.process_poses method. The description of this method says "processes the 1x12 raw pose from dataset by aligning and then normalizing".

So to verify, by raw_pose do you mean the extrinsic matrix (world to camera) or camera to world matrix (which is the ground truth in 7 Scene dataset)?

Thank you for clarifying.

**Comment by samarth-robo** [comment]
> So to verify, by raw_pose do you mean the extrinsic matrix (world to camera) or camera to world matrix (which is the ground truth in 7 Scene dataset)?

@AntiLibrary5 what do you mean by "raw_pose"?

**Comment by samarth-robo** [comment]
@AntiLibrary5 my reply uses notation from this [blog post](https://samarth-robo.github.io/blog/2019/12/29/gtsam_conventions.html) about naming conventions.

You are right that 7Scenes provides camera-to-world matrix. This matrix `wTc` transforms points from the camera coordinate system to the world coordinate system: `wX = wTc @ cX`.

"raw pose" in the docstring of `pose_utils.process_poses()` means this `wTc` matrix. And I also call this matrix as "camera pose" in my earlier comment.

**Comment by AntiLibrary5** [comment]
Hi,
Thank you very much. This is the exact clarification I was looking for. Just a suggestion, consider adding this piece of info in the README for custom dataloaders. I think it would be helpful. But thank you again.

**Comment by samarth-robo** [comment]
> Hi,
> Thank you very much. This is the exact clarification I was looking for. Just a suggestion, consider adding this piece of info in the README for custom dataloaders. I think it would be helpful. But thank you again.

done.
